### PR TITLE
Add sorting to calendar entries in month view

### LINKED ISSUE #1589: [FEATURE] [kalender] Sortierung der WV, Fristen und Termine in der Monatsansicht
derzeitige Sortierung/Darstellung
WV, 
Fristen
Termine, 

oder WV,
Fristen
Termine
(uneinheitlich)

Wunsch: Termine, Fristen, WV

Zweck: die Monatansicht sollte die Verfügbarkeit des Sachbearbeiters auf einen Blick beantworten können. Das geht aktuell nicht, wenn viele WVen und Fristen die Darstellung der Termine verhindern

### PR DESCRIPTION
**Pull request by Bentipa**
Closes #1589

image

Events are now sorted in the order Termin -> Frist -> WV

### CODE REVIEW

**Review by j-dimension — CHANGES_REQUESTED** [review]

File: j-lawyer-client/src/de/costache/calendar/ui/DayContentPanel.java [hunk 1]
```diff
@@ -335,6 +336,23 @@ private void drawBackground(final Graphics2D graphics2d) {
 
     }
 
+
+    private List<CalendarEvent> sortEvents(List<CalendarEvent> events) {
+        events.sort((o1, o2) -> {
+            if (o1.getType().getUniqueKey().equals(o2.getType().getUniqueKey())) {
```
[the remark this answers is not in the document]
Comment (Bentipa) [reply j-lawyer-client/src/de/costache/calendar/ui/DayContentPanel.java:346]: done

File: j-lawyer-client/src/de/costache/calendar/ui/DayContentPanel.java [hunk 2]
```diff
@@ -335,6 +336,23 @@ private void drawBackground(final Graphics2D graphics2d) {
 
     }
 
+
+    private List<CalendarEvent> sortEvents(List<CalendarEvent> events) {
+        events.sort((o1, o2) -> {
+            if (o1.getType().getUniqueKey().equals(o2.getType().getUniqueKey())) {
+                return 0;
+            } else if (o1.getType().getUniqueKey().equals("Termin")) {
```
[the remark this answers is not in the document]
Comment (Bentipa) [reply j-lawyer-client/src/de/costache/calendar/ui/DayContentPanel.java]: done

**Comment by j-dimension** [comment]
@Bentipa thanks for your contributions!

Related to this PR, I noticed an issue with the tooltips being shown for different events. That is due to the sorting taking place on paint, but then in case of mouse events the events are retrieved by their location based on the unsorted list in 

`private CalendarEvent getEvent(final int x, final int y)`

(there are several of them)
So I added the sorting there as well. Might become an issue for databases with millions of events, but I do not know of such installation yet :-)

**Comment by Bentipa** [comment]
@j-dimension Ah yeah that makes sense :) 
Maybe the sorting could also already take place earlier, maybe when adding entries?
This would remove the sorting operation from the render / getEvent methods and maybe reduce their execution count.

**Comment by j-dimension** [comment]
@Bentipa yes, I will look into converting the internal collection into a TreeSet.

**Pull request merged**